src/tests/java/htsjdk/samtools/CRAMIndexQueryTest.java

```diff
@@ -0,0 +1,464 @@
+/*
+ * The MIT License
+ *
+ * Copyright (c) 2016 The Broad Institute
+ *
+ * Permission is hereby granted, free of charge, to any person obtaining a copy
+ * of this software and associated documentation files (the "Software"), to deal
+ * in the Software without restriction, including without limitation the rights
+ * to use, copy, modify, merge, publish, distribute, sublicense, and/or sell
+ * copies of the Software, and to permit persons to whom the Software is
+ * furnished to do so, subject to the following conditions:
+ *
+ * The above copyright notice and this permission notice shall be included in
+ * all copies or substantial portions of the Software.
+ *
+ * THE SOFTWARE IS PROVIDED "AS IS", WITHOUT WARRANTY OF ANY KIND, EXPRESS OR
+ * IMPLIED, INCLUDING BUT NOT LIMITED TO THE WARRANTIES OF MERCHANTABILITY,
+ * FITNESS FOR A PARTICULAR PURPOSE AND NONINFRINGEMENT. IN NO EVENT SHALL THE
+ * AUTHORS OR COPYRIGHT HOLDERS BE LIABLE FOR ANY CLAIM, DAMAGES OR OTHER
+ * LIABILITY, WHETHER IN AN ACTION OF CONTRACT, TORT OR OTHERWISE, ARISING FROM,
+ * OUT OF OR IN CONNECTION WITH THE SOFTWARE OR THE USE OR OTHER DEALINGS IN
+ * THE SOFTWARE.
+ */
+package htsjdk.samtools;
+
+import htsjdk.samtools.util.CloseableIterator;
+import org.testng.Assert;
+import org.testng.annotations.DataProvider;
+import org.testng.annotations.Test;
+
+import java.io.File;
+import java.io.IOException;
+import java.util.function.Function;
+
+/**
+ * Test index query calls against a *SamReader* on a CRAM File, which will use
+ * whatever index format (.bai or .crai converted to .bai) is available for the
+ * target file.
+ */
+public class CRAMIndexQueryTest {
+
+    private static final File TEST_DATA_DIR = new File("testdata/htsjdk/samtools/cram");
+
+    private static final String cramQueryWithBAI = "cramQueryWithBAI.cram";
+    private static final String cramQueryWithCRAI = "cramQueryWithCRAI.cram";
+    private static final String cramQueryReference = "human_g1k_v37.20.21.10M-10M200k.fasta";
+
+    //TODO: this is the bam /bai equivalent of the cram test file, used to validate the query results
+    //private static final String bamQueryWithBAI = "samtools.bam";
```

Comment:
Per my comment in the PR, I'll remove all the BAM comparison test cases in this file on my last squash after review.